nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/Lock.java

```diff
@@ -77,4 +80,36 @@ public boolean getFailed() {
         return failed;
     }
 
+    /**
+     * @since 11.1
+     */
+    @Override
+    public boolean equals(Object obj) {
+        if(obj instanceof Lock) {
```

Comment:
We tend to use `EqualsBuilder` from Apache Commons Lang3. You have two ways to use it:
- with the static method `EqualsBuilder#reflectionEquals` which takes care of everything but may be slow
- with the declared way: you need to instante `EqualsBuilder` and use `append` method as many as you want, generally one call by fields you want to check.

Reply:
I went of the append